src/dlr_pipeline.cc

```diff
@@ -0,0 +1,182 @@
+#include "dlr_pipeline.h"
+
+#include <stdlib.h>
+#include <cstring>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+
+using namespace dlr;
+
+
+void PipelineModel::CheckModelsCompatibility(const DLRModelPtr& pm, const DLRModelPtr& cm,
```

Comment:
Can we split this into two functions "ValidInput" and "ValidateOutputs"

Reply:
This function checks that m1.input matches m0.output